Print_A_Bot/djwifi/management/commands/system_config.py

```diff
@@ -0,0 +1,273 @@
+            wifi_data['psk'] = re.search('\tpsk=(.*)', wpa_out).group(1)
+
+            wpa_entry = '''network={{
+    ssid={ssid}
+    psk={psk}
+
+    proto=RSN
+    key_mgmt=WPA-PSK
+    pairwise=CCMP
+    auth_alg=OPEN
+    priority={priority}
+}}'''
+        else:
+            wpa_entry = '''network={{
+    ssid={ssid}
+    key_mgmt=NONE
+    priority={priority}
+}}'''
+
+        with open(wpa_conf_path, 'a') as f:
+            f.write('\n')
+            f.write(wpa_entry.format(**wifi_data))
+
+    @staticmethod
+    def _cycle_wifi(mode=None):
+        call(['ifdown', settings.WIFI_INTERFACE])
+        if mode is not None:
+            call(['iwconfig', settings.WIFI_INTERFACE, 'mode', mode])
+        call(['ifup', settings.WIFI_INTERFACE])
+
+    # wifi access point #
+    def _setup_wifi_ap(self):
+        context = self._get_ap_context()
+        try:
+            check_output(['ifconfig', context['hostname']])
+            logger.info('wifi ap {} already setup'.format(context['hostname']))
+            return True
+        except CalledProcessError:
+            logger.info('Setting up virtual access point interface')
+        call(['service', 'hostapd', 'stop'])
+        call(['service', 'dnsmasq', 'stop'])
+
+        self._write_system_template('/etc/dnsmasq.conf', 'access_point/dnsmasq.conf')
+        self._write_system_template('/etc/hostapd/hostapd.conf', 'access_point/hostapd.conf', context)
+        self._write_system_template('/etc/network/interfaces', 'access_point/interfaces.conf', context)
+        self._write_system_template('/etc/default/hostapd', 'access_point/default_hostapd.conf', context)
+        self._write_system_template('/etc/dhcpcd.conf', 'access_point/dhcpcd.conf', context)
+        
+        call(['systemctl', 'enable', 'hostapd', ])
+        call(['systemctl', 'enable', 'dnsmasq', ])
+        return True
+
+    def _ap_start(self):
+        logger.info('Starting access point')
+        self._setup_wifi_ap()
+
+        call(['service', 'hostapd', 'start'])
+        call(['service', 'dnsmasq', 'start'])
+
+        logger.info('Access point will be started')
+
+    def _disable_wifi_ap(self):
+        call(['systemctl', 'disable', 'hostapd', ])
+        call(['systemctl', 'disable', 'dnsmasq', ])
+
+        context = self._get_ap_context()
+        self._write_system_template('/etc/network/interfaces', 'interfaces.conf', context)
+        self._write_system_template('/etc/dhcpcd.conf', 'dhcpcd.conf', context)
+
+    def _ap_stop(self):
+        logger.info('Stopping access point')
+        call(['service', 'hostapd', 'stop'])
+        call(['service', 'dnsmasq', 'stop'])
+
+        self._disable_wifi_ap()
+
+        logger.info('Access point disabled')
+
+    # linux system #
+
+    @staticmethod
+    def _reboot():
+        logger.info('Rebooting system')
+        call(['reboot', ])
+
+    @staticmethod
+    def _clean_logs(size=None):
```

Comment:
leftover from an issue with space filling up after running sensors for a few weeks -- seemed like it may be useful